src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/directives/processor_ui_date.js

```diff
@@ -0,0 +1,62 @@
+import uiModules from 'ui/modules';
+import template from '../views/processor_ui_date.html';
+import _ from 'lodash';
+import keysDeep from '../lib/keys_deep';
+import selectableArray from '../lib/selectable_array';
+import '../styles/_processor_ui_date.less';
+
+const app = uiModules.get('kibana');
+
+//scope.processor, scope.pipeline are attached by the process_container.
+app.directive('processorUiDate', function () {
+  return {
+    restrict: 'E',
+    template: template,
+    controller : function ($scope, debounce) {
+      const processor = $scope.processor;
+      const pipeline = $scope.pipeline;
+
+      function consumeNewInputObject() {
+        $scope.fields = keysDeep(processor.inputObject);
+        refreshFieldData();
+      }
+
+      function refreshFieldData() {
+        $scope.fieldData = _.get(processor.inputObject, processor.sourceField);
+      }
+
+      function processorUiChanged() {
+        pipeline.setDirty();
+      }
+
+      function updateFormats() {
+        const selectedFormats = $scope.formats.map((o) => {
+          if (!o.selected) return;
+          return o.title;
+        });
+        processor.formats = _.compact(selectedFormats);
+
+        $scope.customFormatSelected = !_.isUndefined(_.find(processor.formats, (o) => {
+          return o === 'Custom';
+        }));
+        processorUiChanged();
+      }
+      updateFormats = debounce(updateFormats, 200);
```

Comment:
Same comment here as previous usages of debounce. This could lead to confusing scoping issues, it would be better to do:

```
const updateFormats = debounce(function () {
...
}, 200);
```